### PR TITLE
Update to document new Coastlines paper and use recently added Dask and Geopandas functionality

### PR DESCRIPTION
**Pull request by robbibt**
### Proposed changes
This PR makes some minor updates to a range of notebooks to document the new Coastlines paper and take advantage of recently added functionality:

- Add links to the Coastlines paper in relevant notebooks
- Remove the `map_shapefile` function and replace it with the new `.explore` functionality in `geopandas` that does the same thing. Also added a `FutureWarning` to `map_shapefile` so we can remove that func in the future
- Updated several notebooks to output the Dask client as an object so that it can be shut down at the end of the analysis
- Loop through Multi-part geometries using `.geoms` for compatibility with future versions of Shapely

### CODE REVIEW

File: Tools/dea_tools/spatial.py [hunk 1]
```diff
@@ -704,7 +704,7 @@ def contours_to_arrays(gdf, col):
 
         try:
             coords = np.concatenate([np.vstack(x.coords.xy).T 
-                                     for x in gdf.iloc[i].geometry])
+                                     for x in gdf.iloc[i].geometry.geoms])
```
Comment (MatthewJA) [inline Tools/dea_tools/spatial.py:707]: Why?

Comment (MatthewJA) [reply Tools/dea_tools/spatial.py:707]: I mean I don't doubt this works - but why are we changing it?

Comment (robbibt) [reply Tools/dea_tools/spatial.py:707]: Yep, sorry - forgot to mention this above. Shapely is deprecating looping through Multi* features in Shapely version 2.0, see 1.3.2/Page 80 here: https://buildmedia.readthedocs.org/media/pdf/shapely/latest/shapely.pdf.

The suggested update is to access the parts using `.geoms` instead

**Comment by MatthewJA** [comment]
Just a few complaints and questions @robbibt

**Comment by robbibt** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/887/discussion/) on ReviewNB
> 
> _MatthewJA commented on 2021-10-21T03:39:31Z_ ----------------------------------------------------------------
> 
> If we always zoom into Bussleton what's the point of the interactive map? Maybe we could have a static example for Bussleton...? Or some other way to do this?
> 
> edit: oh I see. Can we auto-zoom somehow? Surely

The text prompt in the markdown was just to avoid users not realising that they could use the map to zoom in - Bussleton is a nice example so even if they don't explore the rest of the map they will get the concept. I originally included code to auto-zoom there, but decided it added unnecessary complexity.

**Comment by robbibt** [comment]
<p>Good catch, that one has been there for a while</p>

---

 <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/887/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by robbibt** [comment]
<p>Will try - it comes up all the time these days though. Fingers crossed for a clean run!</p>

---

 <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/887/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by robbibt** [comment]
<p>Will update!</p>

---

 <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/887/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by robbibt** [comment]
<p>The text prompt in the markdown was just to avoid users not realising that they could use the map to zoom in - Bussleton is a nice example so even if they don't explore the rest of the map they will get the concept. I originally included code to auto-zoom there, but decided it added unnecessary complexity.</p>

---

 <a href='https://app.reviewnb.com/GeoscienceAustralia/dea-notebooks/pull/887/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by robbibt** [comment]
@MatthewJA Fixed up the date, the spelling mistake and got rid of the mutex error in the Coastlines notebook. I re-ran the Dask one five times and wouldn't get the mutex to disappear on any run, so that will unfortunately have to stay as it is for the time being :(

**Review by MatthewJA — APPROVED** [review]
LGTM then!

**Pull request merged**